web-frontend/src/pages/TransferCoins/ReceiveCoins/ReceiveCoins.tsx

```diff
@@ -23,19 +24,47 @@ function ReceiveCoins() {
   const formattedAddress = maskAddress(address);
   const VITE_BASE_APP = import.meta.env.VITE_BASE_APP;
   const baseURL = window.location.origin;
-
+  const [modal, setModal] = useState(false);
   const presetURL = `${baseURL}${VITE_BASE_APP}/send-redirect/?to=${address}`;
+  const [url, setURL] = useState(presetURL);
+  const [provider, setProvider] = useState('');
+  const [amount, setAmount] = useState('');
+
+  console.log('url', url);
+
+  const handleGenerate = () => {
+    const amountArg = amount ? `&amount=${amount}` : '';
+    const providerArg = provider ? `&provider=${provider}` : '';
+    const newURL = presetURL + amountArg + providerArg;
+    setURL(newURL);
+  };
+
+  const modalArgs = {
+    account,
+    modal,
+    url,
+    setModal,
+    setAmount,
+    setProvider,
+    handleGenerate,
+  };
+
+  const copyContentArgs = {
+    content: address,
+    formattedContent: formattedAddress,
+  };
```

Comment:
I think it's better to have the props for a component where you insert that component, it improve readability